tests/Integration/inc/classes/ImagifyUser/TestGetPercentConsumedQuota.php

```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Imagify\tests\Integration\inc\classes\ImagifyUser;
+
+use Brain\Monkey\Functions;
+use Imagify\tests\Integration\TestCase;
+
+use Imagify_Data;
+use Imagify_User;
+use WP_Error;
+
+/**
+ * Tests for Imagify_User->get_percent_consumed_quota().
+ *
+ * @covers Imagify_User::get_percent_consumed_quota
+ * @group  ImagifyAPI
+ */
+class Test_GetPercentConsumedQuota extends TestCase {
+
+	/**
+	 * Test Imagify_User->get_percent_consumed_quota() should return 0 when couldn’t fetch user account data.
+	 */
+	public function testShouldReturnZeroWhenCouldNotFetchUserData() {
+		$wp_error = new WP_Error( 'error_id', 'Error Message' );
+
+		Functions\when( 'get_imagify_user' )->justReturn( $wp_error );
+		Functions\expect( 'imagify_round_half_five' )->never();
+
+		$this->assertSame( ( new Imagify_User() )->get_percent_consumed_quota(), 0 );
```

Comment:
Instead mocking the functions, let them run. How? Before instantiating the user, set up the conditions for the API not to find and return the user's data. 

Think about what would cause this state, such as the user doesn't have an account. Once you know what that is, then look for a way to hook into `Imagify::get_user()` to trigger that state when calling the API, e.g. pass it an invalid user credentials.

Another tip: Move the `static $done` to be a static property on the class. Then use  `$this-set_reflective_property()` to reset it after each  test.

```php
// Set up the "couldn't fetch user account data" condition here.

$user = new Imagify_User();
$this->assertInstanceOf( 'WP_Error', $user->get_error() );
$this->assertSame( 0, $user-> get_percent_consumed_quota() );
```